src/components/auth_row/index.ts

```diff
@@ -0,0 +1,23 @@
+import template from './auth_row.hbs';
+import Block from '../../utils/Block';
+
+export interface AuthProps {
+    div_class?: string;
+    name: string;
+    type: string;
```

Comment:
Полагаю, что type может принимать ограниченное количество значений. Если это так, то лучше ограничить тип, используя discriminated union 
```suggestion
    type: 'type1' | 'type2' и тд;
```